modules/client-handler/src/main/java/org/apache/ignite/client/handler/JdbcQueryEventHandlerImpl.java

```diff
@@ -287,6 +300,26 @@ public CompletableFuture<JdbcMetaPrimaryKeysResult> primaryKeysMetaAsync(JdbcMet
         return meta.getPrimaryKeys(req.schemaName(), req.tableName()).thenApply(JdbcMetaPrimaryKeysResult::new);
     }
 
+    /** {@inheritDoc} */
+    @Override
+    public CompletableFuture<JdbcFinishTxResult> finishTxAsync(long connectionId, boolean commit) {
+        JdbcConnectionContext connectionContext;
+
+        try {
+            connectionContext = resources.get(connectionId).get(JdbcConnectionContext.class);
+        } catch (IgniteInternalCheckedException exception) {
+            return CompletableFuture.completedFuture(new JdbcFinishTxResult(Response.STATUS_FAILED, "Connection is broken"));
+        }
+
+        return connectionContext.finishTransactionAsync(commit).handle((ignored, t) -> {
```

Comment:
Also, I'd split methods with flags (like connectionContext.finishTransactionAsync) into commitAsync and rollbackAsync to be. more specific.

Reply:
I tried it and I like it better the way it is now (less code/duplicate code), but if you insist I can split it.